src/rgw/rgw_civetweb.cc

```diff
@@ -51,11 +51,13 @@ RGWCivetWeb::RGWCivetWeb(mg_connection* const conn)
 
 size_t RGWCivetWeb::read_data(char *buf, size_t len)
 {
-  int c, ret;
+  int ret;
+  size_t c;
+
   if (got_eof_on_read) {
     return 0;
   }
-  for (c = 0; c < len; c += ret) {
+  for (c = 0; c < len; c += (size_t)ret) {
```

Comment:
Here `len` is the cause. So better do `for (c = 0; c < static_cast<int>(len); c += ret) {` ?

Reply:
First, I thought the same. But, I changed it to this because:-
1) The function is returning value of type size_t . 
2) c is always non-negative so it's better to change it to size_t.

Reply:
@joscollin @mdw-at-linuxbox @cbodley Please look at this PR #21105